core/src/main/scala/org/apache/spark/SSLOptions.scala

```diff
@@ -179,9 +185,11 @@ private[spark] object SSLOptions extends Logging {
         .orElse(defaults.flatMap(_.keyStore))
 
     val keyStorePassword = conf.getWithSubstitution(s"$ns.keyStorePassword")
+        .orElse(Option(hadoopConf.getPassword(s"$ns.keyStorePassword")).map(new String(_)))
```

Comment:
Needs charset (also in others).

Reply:
`new String` takes a charset. (In fact the constructor you're calling should be deprecated...)

Reply:
Hi @vanzin , I checked jdk8 doc again, I don't find a String constructor which takes both char array and charset as parameters.

Reply:
Oh, my bad, that's a char array, not a byte array. All is good then.